Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityController.cs

```diff
@@ -61,443 +61,29 @@ public WindowsMixedRealityController(TrackingState trackingState, Handedness con
-                        if (interactionMapping.Changed)
-                        {
-                            // Raise input system Event if it enabled
-                            if (interactionSourceState.touchpadTouched)
-                            {
-                                InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
-                            }
-                            else
-                            {
-                                InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
-                            }
-                        }
-                        break;
-                    }
-                case DeviceInputType.TouchpadPress:
-                    {
-                        //Update the interaction data source
-                        interactionMapping.BoolData = interactionSourceState.touchpadPressed;
-
-                        // If our value changed raise it.
-                        if (interactionMapping.Changed)
-                        {
-                            // Raise input system Event if it enabled
-                            if (interactionSourceState.touchpadPressed)
-                            {
-                                InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
-                            }
-                            else
-                            {
-                                InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
-                            }
-                        }
-                        break;
-                    }
-                case DeviceInputType.Touchpad:
-                    {
-                        // Update the interaction data source
-                        interactionMapping.Vector2Data = interactionSourceState.touchpadPosition;
-
-                        // If our value changed raise it.
-                        if (interactionMapping.Changed)
-                        {
-                            // Raise input system Event if it enabled
-                            InputSystem?.RaisePositionInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionSourceState.touchpadPosition);
-                        }
-                        break;
-                    }
-            }
-        }
-
-        /// <summary>
-        /// Update the Thumbstick input from the device
-        /// </summary>
-        /// <param name="interactionSourceState">The InteractionSourceState retrieved from the platform</param>
-        /// <param name="interactionMapping"></param>
-        private void UpdateThumbStickData(InteractionSourceState interactionSourceState, MixedRealityInteractionMapping interactionMapping)
```

Comment:
It was a little surprising to see the thumbstick and touchpad methods moved into the base class, since it feels like those are specific to the motion controller.

Reply:
Same with trigger and menu buttons.

Reply:
Yeah, I waffled a bit on what should go where. I ended up sticking everything from the `InteractionSourceState` in the base class. My rational for some of them was "like we started sending data for hands over the `grasped` state, what if we start sending something for, say, menu in the future".

I agree though, in the current world, I'll move motion controller specific buttons to the controller class explicitly.

Reply:
Updated